safenode/src/protocol/online_transfers/error.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use sn_dbc::Error as DbcError;
+
+use thiserror::Error;
+
+pub(crate) type Result<T> = std::result::Result<T, Error>;
+
+/// Error type returned by the API
+#[derive(Debug, Error)]
+#[allow(clippy::large_enum_variant)]
+#[non_exhaustive]
+pub enum Error {
```

Comment:
better rename this to `OnlineTransferError` and the one for offline to be `OfflineTransferError`
bearing the same enum name and having most entries the same naming for both may cause trouble/confusion for tracing.

Reply:
These are merged into one in the follow up PR.

> NB 1: Some duplication of types in the online transfers mod. This will be improved upon in the followup PRs.